azure/services/asyncpoller/asyncpoller.go

```diff
@@ -73,34 +73,40 @@ func (s *Service[C, D]) CreateOrUpdateResource(ctx context.Context, spec azure.R
+	// The polling implemented by the SDK does not use parameters when a resume token exists.
+	var parameters interface{}
+	if resumeToken == "" {
+		// Get the resource if it already exists, and use it to construct the desired resource parameters.
+		var existingResource interface{}
+		if existing, err := s.Creator.Get(ctx, spec); err != nil && !azure.ResourceNotFound(err) {
+			errWrapped := errors.Wrapf(err, "failed to get existing resource %s/%s (service: %s)", rgName, resourceName, serviceName)
+			return nil, azure.WithTransientError(errWrapped, getRetryAfterFromError(err))
+		} else if err == nil {
+			existingResource = existing
+			log.V(2).Info("successfully got existing resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+		}
 
-	// Construct parameters using the resource spec and information from the existing resource, if there is one.
-	parameters, err := spec.Parameters(ctx, existingResource)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get desired parameters for resource %s/%s (service: %s)", rgName, resourceName, serviceName)
-	} else if parameters == nil {
-		// Nothing to do, don't create or update the resource and return the existing resource.
-		log.V(2).Info("resource up to date", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
-		return existingResource, nil
-	}
+		// Construct parameters using the resource spec and information from the existing resource, if there is one.
+		parameters, err = spec.Parameters(ctx, existingResource)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get desired parameters for resource %s/%s (service: %s)", rgName, resourceName, serviceName)
+		} else if parameters == nil {
+			// Nothing to do, don't create or update the resource and return the existing resource.
+			log.V(2).Info("resource up to date", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+			return existingResource, nil
+		}
 
-	// Create or update the resource with the desired parameters.
-	logMessageVerbPrefix := "creat"
-	if existingResource != nil {
-		logMessageVerbPrefix = "updat"
+		// Create or update the resource with the desired parameters.
+		logMessageVerbPrefix := "creat"
+		if existingResource != nil {
+			logMessageVerbPrefix = "updat"
+		}
+		log.V(2).Info(fmt.Sprintf("%sing resource", logMessageVerbPrefix), "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
 	}
-	log.V(2).Info(fmt.Sprintf("%sing resource", logMessageVerbPrefix), "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+
 	result, poller, err := s.Creator.CreateOrUpdateAsync(ctx, spec, resumeToken, parameters)
-	errWrapped := errors.Wrapf(err, fmt.Sprintf("failed to %se resource %s/%s (service: %s)", logMessageVerbPrefix, rgName, resourceName, serviceName))
+	errWrapped := errors.Wrapf(err, "failed to create or update resource %s/%s (service: %s)", rgName, resourceName, serviceName)
```

Comment:
This wording change is not just me being lazy, but I don't think we can reliably know which of "create" or "update" is really happening here since we no longer always do a GET.